testsuites/listener/shared/client_sg/test_replication.py

```diff
@@ -1224,3 +1420,112 @@ def test_replication_with_session_cookie_short_ttl(setup_client_syncgateway_test
 
     # Delete the session
     client.delete_session(url=sg_admin_url, db=sg_db, user_name="user_1", session_id=session_id)
+
+
+@pytest.mark.listener
+@pytest.mark.syncgateway
+@pytest.mark.replication
+def test_replication_attachments_survive_channel_removal(setup_client_syncgateway_test):
+    """Regression test for https://github.com/couchbase/couchbase-lite-net/issues/910
+    1. SyncGateway Config with One user added (e.g. user1 / 1234) who has access to a limited
+       set of channels (e.g. ["user_channel"]) and GUEST disabled
+    2. Start authenticated continuous pull replication
+    3. Add a document that gets assigned to the user's channel, and verify that it is pulled
+    4. Stop the pull replication started in (2)
+    5. Add a few arbitrary edits to the document created in (3), including adding at least
+       one attachment
+    6. Add an edit to the document created in (3) that removes it from the user's channel
+    7. Repeat step (2)
+    8. Add an edit to the document created in (3) that keep the attachments intact, as well
+       as adds the document back into the user's channel
+    9. Verify that the document and attachment are both present in the local DB
+    """
+
+    # Here is the general idea, tweak and remove this comment
```

Comment:
What are we tweaking and removing?

Reply:
oops, forgot to remove the comment from Jim